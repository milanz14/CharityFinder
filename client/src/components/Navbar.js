import React from "react";
import { Switch, Route, NavLink } from "react-router-dom";
import "../styles/Navbar.css";

const Navbar = () => {
    return (
        <div>
            <nav className="Navbar">
                <NavLink exact className="nav-link" to="/">
                    Home
                </NavLink>
                <NavLink exact className="nav-link" to="/organizations">
                    All Organizations
                </NavLink>
                <NavLink exact className="nav-link" to="/categories">
                    Search
                </NavLink>
                <NavLink exact className="nav-link" to="/register">
                    Register
                </NavLink>
            </nav>
        </div>
    );
};

export default Navbar;
