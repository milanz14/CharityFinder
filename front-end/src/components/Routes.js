import { Route, Switch, Redirect } from "react-router-dom";
import Organizations from "./Organizations";
import Home from "./Home";
import CategorySearch from "./CategorySearch";
import LocationSearch from "./LocationSearch";

const Routes = () => {
    return (
        <Switch>
            <Route exact path="/">
                <Home />
            </Route>
            <Route exact path="/organizations">
                <Organizations />
            </Route>
            <Route exact path="/categories">
                <CategorySearch />
            </Route>
            <Route exact path="/locations">
                <LocationSearch />
            </Route>
        </Switch>
    );
};

export default Routes;
