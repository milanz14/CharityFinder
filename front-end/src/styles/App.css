.App {
    text-align: center;
    background-color: #97caef;
}
