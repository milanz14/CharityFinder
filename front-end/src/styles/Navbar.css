.Navbar {
    display: flex;
    justify-content: space-around;
    background-color: #55bcc9;
    padding: 30px;
}

.Navbar a {
    color: cornsilk;
    transition: all 200ms ease-in;
}

.Navbar a:hover {
    transform: scale(0.95);
    color: black;
}

.Navbar a.active {
    color: white;
    text-decoration: underline;
}

.nav-link {
    color: white;
}
