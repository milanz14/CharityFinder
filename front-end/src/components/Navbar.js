import React from "react";
import { Switch, Route, NavLink } from "react-router-dom";
import "../styles/Navbar.css";

const Navbar = () => {
    return (
        <div>
            <nav className="Navbar">
                <NavLink exact activeClassName="Navbar-active" to="/">
                    Home
                </NavLink>
                <NavLink exact activeClassName="Navbar-active" to="/all">
                    All Charities
                </NavLink>
                <NavLink exact activeClassName="Navbar-active" to="/categories">
                    Category Search
                </NavLink>
                <NavLink exact activeClassName="Navbar-active" to="/locations">
                    Location Seach
                </NavLink>
                <NavLink exact activeClassName="Navbar-active" to="/register">
                    Register
                </NavLink>
            </nav>
        </div>
    );
};

export default Navbar;
