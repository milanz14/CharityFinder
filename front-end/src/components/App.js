import Routes from "./Routes";
import Navbar from "./Navbar";
import SearchRouteContainer from "./SearchRouteContainer";
import Organizations from "./Organizations";
import "../styles/App.css";

function App() {
    return (
        <div className="App">
            <Navbar />
            <Routes />
            {/* <SearchRouteContainer /> */}
            <Organizations />
        </div>
    );
}

export default App;
