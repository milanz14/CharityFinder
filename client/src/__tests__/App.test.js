import { render, screen } from "@testing-library/react";
import App from "../components/App";
import React from "react";
import Navbar from "../components/Navbar";
import Routes from "../components/Routes";

test("it renders without crashing", () => {
    render(<App />);
});

test("it matches the snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
});
