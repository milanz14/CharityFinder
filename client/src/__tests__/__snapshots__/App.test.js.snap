// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <nav
        class="Navbar"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="/organizations"
        >
          All Organizations
        </a>
        <a
          class="nav-link"
          href="/categories"
        >
          Search
        </a>
        <a
          class="nav-link"
          href="/register"
        >
          Register
        </a>
      </nav>
    </div>
    <div
      class="home"
    >
      <h1
        class="home-title"
      >
        Feeling Charitable?
      </h1>
      <a
        class="home-btn"
        href="/organizations"
      >
        Let's Get Started
      </a>
    </div>
  </div>
</DocumentFragment>
`;
